lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -383,36 +383,43 @@ struct DifferentiationInvoker {
   void print(llvm::raw_ostream &os) const;
 };
 
-/// Information about the VJP function produced during VJP generation, e.g.
-/// mappings from original values to corresponding values in the pullback
+/// Information about the VJP/JVP function produced during VJP/JVP generation,
+/// e.g. mappings from original values to corresponding values in the linear map
 /// struct.
 ///
-/// A pullback struct is an aggregate value containing pullbacks checkpointed
-/// during the VJP computation. Pullback structs are generated for every
-/// original function during VJP generation. Pullback struct values are
-/// constructed by VJP functions and consumed by pullback functions.
-class PullbackInfo {
+/// A linear map struct is an aggregate value containing linear maps checkpointed
+/// during the VJP/JVP computation. Linear map structs are generated for every
+/// original function during VJP/JVP generation. Linear map struct values are
+/// constructed by VJP/JVP functions and consumed by pullback/differential
+/// functions.
+class LinearMapInfo {
 private:
+  /// The linear map kind.
+  AutoDiffAssociatedFunctionKind kind;
+
   /// The original function.
   SILFunction *const original;
 
-  /// Mapping from original basic blocks to pullback structs.
-  DenseMap<SILBasicBlock *, StructDecl *> pullbackStructs;
+  /// Mapping from original basic blocks to linear map structs.
+  DenseMap<SILBasicBlock *, StructDecl *> linearMapStructs;
 
-  /// Mapping from original basic blocks to predecessor enums.
-  DenseMap<SILBasicBlock *, EnumDecl *> predecessorEnums;
+  /// Mapping from original basic blocks to linear map enums. If 'kind' is 'VJP',
+  /// then it's the predecessor enums. If 'kind' is 'JVP', then it's the
+  /// successor enums.
+  DenseMap<SILBasicBlock *, EnumDecl *> linearMapEnums;
 
   /// Mapping from `apply` and `struct_extract` instructions in the original
-  /// function to the corresponding pullback declaration in the pullback struct.
-  DenseMap<SILInstruction *, VarDecl *> pullbackValueMap;
+  /// function to the corresponding linear map declaration in the linear map
+  /// struct.
+  DenseMap<SILInstruction *, VarDecl *> linearMapValueMap;
```

Comment:
Note: this is currently necessary only for VJP/pullback generation and not for JVP/differential generation.
This will become necessary again with JVP -> differential -> pullback (transposed differential) generation.